### PR TITLE
ci: add lint workflow

### PR DESCRIPTION
**Pull request by asafMasa**

### CODE REVIEW

File: projectTemplates/ts-server/index.ts [hunk 1]
```diff
@@ -7,12 +7,16 @@ import { getApp } from './src/app';
...
   const probe = container.resolve(Probe);
   await probe.start(app, port);
   probe.readyFlag = true;
 }
 
-main();
+main()
+    .catch(err => console.log(`main function unhandled exception ${JSON.stringify(err)}`))
+    .then(() => console.log('main ended'))
+    .catch(() => 'obligatory catch');
```
Comment (syncush) [inline projectTemplates/ts-server/index.ts]:
```js
JSON.stringify(err)
```
will result in {}.
print stack and message properties of error

File: projectTemplates/ts-server/src/middleware/ErrorHandler.ts [hunk 2]
```diff
@@ -23,7 +25,7 @@ export class ErrorHandler {
           `${req.method} request to ${req.originalUrl}  has failed with error: ${err.message}`
         );
         if (process.env.NODE_ENV === 'development') {
-          res.status(500).json({
+          res.status(defaultServerErrorCode).json({
```
Comment (syncush) [inline projectTemplates/ts-server/src/middleware/ErrorHandler.ts]: use the `HttpStatus` module

**Review by syncush — CHANGES_REQUESTED** [review]

File: projectTemplates/ts-server/index.ts [hunk 3]
```diff
@@ -6,13 +6,19 @@ import { container } from 'tsyringe';
...
+	const app = await getApp();
+	const probe = container.resolve(Probe);
+	await probe.start(app, port);
+	probe.readyFlag = true;
 }
 
-main();
+main()
+	.catch((error: Error) => {
+		console.log(`main function unhandled exception message: ${ error.message }, stack: ${ error.stack ?? '' }`);
```
Comment (syncush) [inline projectTemplates/ts-server/index.ts]: not supported in node `v12.18.3`

**Review by syncush — CHANGES_REQUESTED** [review]

File: projectTemplates/ts-server/src/middleware/ErrorHandler.ts (hunk 2 above)
Comment (asafMasa) [reply projectTemplates/ts-server/src/middleware/ErrorHandler.ts]: fixed

File: projectTemplates/ts-server/index.ts (hunk 1 above)
Comment (asafMasa) [reply projectTemplates/ts-server/index.ts]: fixed

File: projectTemplates/ts-server/index.ts (hunk 3 above)
Comment (asafMasa) [reply projectTemplates/ts-server/index.ts]: fixed

File: .github/workflows/lintJsProject.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+  
+name: Lint
+
+on: [push, pull_request]
```
Comment (syncush) [inline .github/workflows/lintJsProject.yml]: push is redundant

File: .github/workflows/lintTsProject.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+  
+name: Lint
+
+on: [push, pull_request]
```
Comment (syncush) [inline .github/workflows/lintTsProject.yml]: same

File: projectTemplates/js-server/package.json [hunk 6]
```diff
@@ -43,6 +44,7 @@
     "eslint-plugin-flowtype": "^4.7.0",
     "eslint-plugin-import": "^2.22.0",
     "eslint-plugin-jsx-a11y": "^6.3.1",
+    "eslint-plugin-markdown": "^2.0.0-rc.0",
```
Comment (syncush) [inline projectTemplates/js-server/package.json]: is there a version that is not rc? we should stick to stable

**Review by syncush — CHANGES_REQUESTED** [review]

File: .github/workflows/lintJsProject.yml (hunk 4 above)
Comment (asafMasa) [reply .github/workflows/lintJsProject.yml]: fixed

File: .github/workflows/lintTsProject.yml (hunk 5 above)
Comment (asafMasa) [reply .github/workflows/lintTsProject.yml]: fixed

File: projectTemplates/js-server/package.json (hunk 6 above)
Comment (asafMasa) [reply projectTemplates/js-server/package.json]: changed to latest 1.0.2

File: projectTemplates/js-server/.eslintrc.json [hunk 7]
```diff
@@ -13,7 +15,7 @@
     "space-before-function-paren": "off",
     "indent": [
       "error",
-      2
+      4
```
Comment (syncush) [inline projectTemplates/js-server/.eslintrc.json]: I think 2 is better lets set a vote, or we can set it to 2 but the people in your team can change it later to 4 or the vice versa, I don't really care.

Comment (asafMasa) [reply projectTemplates/js-server/.eslintrc.json]: changed

**Review by syncush — APPROVED** [review]

**Pull request merged**